src/api-v1.js

```diff
@@ -0,0 +1,280 @@
+let debug = require('debug')('cloud-mirror:api-v1');
+let base = require('taskcluster-base');
+let taskcluster = require('taskcluster-client');
+let _ = require('lodash');
+let delayer = require('./delayer');
+
+let GENERIC_ID_PATTERN = /^[a-zA-Z0-9-_]{1,22}$/;
+
+let api = new base.API({
+  title: 'Cloud Mirror API',
+  description: 'Service to duplicate URLs from various cloud providers',
+  schemaPrefix: 'http://schemas.taskcluster.net/s3-distribute/v1/',
+  params: {
+    taskId: GENERIC_ID_PATTERN,
+    runId: GENERIC_ID_PATTERN,
+    name: GENERIC_ID_PATTERN,
+  },
+});
+
+module.exports = api;
+
+api.declare({
+  method: 'get',
+  // Note that the Error parameter is only here to check for improperly
+  // URL-Encoded URLs.  This would likely cause errors if this API were
+  // included in the taskcluster-client* packages.  Maybe we should have a
+  // parameter to api.declare that lets us ignore certain endpoints from
+  // generated API-References.  A value of '', e.g "/redirect/s/r/u/" would be
+  // fine, it just has to evaluate as falsy
+  route: '/redirect/:service/:region/:url/:error?',
+  name: 'redirect', 
+  //deferAuth: false,
+  //scopes: [],
+  title: "Redirect to backing cache", 
+  description: [
+    'Redirect to the copy of :url in :region.  If there is',
+    'no copy of the file in that region, submit a request to',
+    'backend process to copy into that region and wait to respond',
+    'here until that happens',
+    '',
+    'NOTE: URL parameter must be URL Encoded!',
+    '',
+    'NOTE: If using this with an api-reference consuming client',
+    'you will need to pass error as an empty string!',
+  ].join('\n'),
+}, async function (req, res) {
+  let url = req.params.url;
+  let region = req.params.region;
+  let service = req.params.service;
+  let error = req.params.error;
+  if (error) {
+    return res.status(400).json({
+      msg: 'URL Must be URL encoded!',
+    });
+  }
+
+  let logthingy = `${url} in ${service}/${region}`;
+  debug(`Attempting to redirect to ${logthingy}`);
+
+  if (service.toLowerCase() === 's3') {
+    if (this.s3backends[region]) {
+      let backend = this.s3backends[region];
+      try {
+        await backend.validateInputURL(url);
```

Comment:
How is this feasible.. .You can't make a request to source url for every request... You shouldn't be hitting the source url here. Only backend should hit the source url... Here you should hit the cache, then the expected cachedUrl, and then send request... You got that right in `getBackendUrl`.